mart/datamodules/coco.py

```diff
@@ -69,6 +69,23 @@ def _load_target(self, id: int) -> List[Any]:
 
         return {"image_id": id, "file_name": file_name, "annotations": annotations}
 
+    def __getitem__(self, index: int):
+        """Override __getitem__() to dictionarize input for multi-modality datasets.
+
+        This runs after _load_image() and transforms(), while transforms() typically converts
+        images to tensors.
+        """
+
+        image, target_dict = super().__getitem__(index)
+
+        # Convert multi-modal input to a dictionary.
+        if self.modalities is not None:
```

Comment:
Maybe we should add a parameter like `return_dict=True` to `CocoDetection`?

Reply:
Then you fix this:
https://github.com/IntelLabs/MART/blob/bafd3a54974c95cc6d90fabfc768b4378c73a461/mart/models/modular.py#L102

Reply:
Batch is formed in the collate function. I think it might be better to merge input and target into a `dict` in `collate_fn()` so that we don't have to change every dataset.

Reply:
Makes sense!